src/inmanta/server/agentmanager.py

```diff
@@ -921,7 +922,10 @@ async def request_parameter(self, env_id: uuid.UUID, resource_id: ResourceIdStr)
             if res is None:
                 return 404, {"message": "The resource has no recent version."}
 
-            if res.status in UNDEPLOYABLE_NAMES:
+            rps = await data.ResourcePersistentState.get_one(environment=env_id, resource_id=resource_id)
```

Comment:
This should probably be transactional. @wouterdb wdyt?

Reply:
Good catch, I don't think it is. But I've got a plan so cunning you could put a tail on it and call it a weasel (can't help it, the word "cunning" only has one association for me).

On a more serious note: I think we should fix it, and it should be as easy (off the top of my head) as moving the check for blocked from [`deploy_start`](https://github.com/inmanta/inmanta-core/blob/2fca03c76a68b9fc4fb625169246f4416d7e47c8/src/inmanta/deploy/scheduler.py#L1119) lower in the call stack to [`_get_resource_intent`](https://github.com/inmanta/inmanta-core/blob/2fca03c76a68b9fc4fb625169246f4416d7e47c8/src/inmanta/deploy/scheduler.py#L1096), so that `get_resource_version_intent` reaches it as well.